lib/facility.rb

```diff
@@ -1,14 +1,189 @@
+        false
+      end
+    else
+      false
+    end
+  end
+
+  def administer_road_test(name)
+    if services.find {|service| service.include?('Road Test')}
+      if name.license_data[:written] == true
+        name.license_data[:license] = true
+        true
+      end
+    else
+      false
+    end
+  end
+
+  def renew_drivers_license(name)
+    if services.find {|service| service.include?('Renew License')}
+      if name.license_data[:written] == true && name.license_data[:license] == true
+        name.license_data[:renewed] = true
+      elsif name.license_data[:written] == false || name.license_data[:license] == false
+        name.license_data[:renewed] = false
+      end
+    else
+      false
+    end
+  end
+
+  def create_facility(list) #filter by something more permanent
+    nf_arr = []
+    if list.count == 5 || list.find {|omv| omv[:state] == "CO"} #CO
+      list.each do |omv|
+        new_omv = {
+          name: omv[:dmv_office],
+          address: omv[:address_li], #+ ", " + omv[:address__1],
+          phone: omv[:phone]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    elsif list.count == 179 || list.find {|omv| omv[:state] == "MO"} #MO
+      list.each do |omv|
+        new_omv = {
+          name: omv[:name],
+          address: omv[:address1],
+          phone: omv[:phone]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    elsif list.count == 172 || list.find {|omv| omv[:state] == "NY"} #NY
+      list.each do |omv|
+        new_omv = {
+          name: omv[:office_name],
+          address: omv[:street_address_line_1],
+          phone: omv[:public_phone_number]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    else
+      p "UNKNOWN DATA SET"
+    end
+    return nf_arr
+  end
+
+  def hours(source)
+    if source.find {|omv| omv[:state] == "CO"} #CO
+      hour_array = []
+      source.each do |omv|
+        office_hours = {
+          omv[:dmv_office] => omv[:hours]
+        }
+        hour_array << office_hours
+      end
+      hour_array
+    elsif source.find {|omv| omv[:state] == "NY"} #NY
+      hour_array = []
+      source.each do |omv|
+        office_hours = {
+          omv[:office_name] =>
+          [omv[:monday_beginning_hours],
+          omv[:monday_ending_hours],
+          omv[:tuesday_beginning_hours],
+          omv[:tuesday_ending_hours],
+          omv[:wednesday_beginning_hours],
+          omv[:wednesday_ending_hours],
+          omv[:thursday_beginning_hours],
+          omv[:thursday_ending_hours],
+          omv[:friday_beginning_hours],
+          omv[:friday_ending_hours]]
+        }
+        hour_array << office_hours
+      end
+      hour_array
+    elsif source.find {|omv| omv[:state] == "MO"} #MO
+      hour_array = []
+      source.each do |omv|
+        office_hours = {
+          omv[:name] => omv[:daysopen]
+        }
+        hour_array << office_hours
+      end
+      hour_array
+    end
+  end
```

Comment:
You could also update your `create_facilities` method to pull hours out of the data and pass that information along to the FAcility class. Then, you would need to update your facility class to pull out an attribute for hours, and have a way to read that data (`attr_reader`)

Reply:
I like the above approach because then you could work with FAcility objects to find out the hour information rather than just raw Hash data.